### ISSUE TITLE
Bring launch sequencing in line with DAP sec

### ISSUE THREAD

**Issue by JustinGrote** [PowerShell/PowerShellEditorServices#2191]
### Prerequisites

- [X] I have written a descriptive issue title.
- [X] I have searched all [issues](https://github.com/PowerShell/PowerShellEditorServices/issues?q=is%3Aissue) to ensure it has not already been requested.

### Summary

Currently, our DAP conversation doesn't follow the spec explicitly in terms of conversation.

What is supposed to happen:
1. Initialize is sent and server responds, then responds with capabilities
1. Launch is sent by client after receiving capabilities. **should be held and deferred until configurationDone**
1. Server sends initialized event (preferably when ready to receive launch requests)
3. Client does config after initialized 
4. Configuration is done (breakpoints etc.)
5. launch request is now responded to after configuration is completed

In actuality, we respond to the launch request immediately before configuration
![image](https://github.com/user-attachments/assets/6d9ebc27-922a-4c0b-a6bb-b674ba0695fd)

The spec states:
```
Once the debug adapter is ready to receive configuration from the client, it sends an initialized event to the client. As described above, the client sends zero or more configuration-related requests before sending a configurationDone request.

After the response to configurationDone is sent, the debug adapter may respond to the launch or attach request, and then the debug session has started.
```

### Proposed Design

We need to hold the launch response until after configurationDone is completed. Fixing this would likely be part of a larger refactor where the jobs are started within configurationDone, rather than configurationDone triggering an event or otherwise to start/release any pending launch messages.

**Comment by jborean93** [comment]
Moving the sequence here to align with how a normal DAP client would work would help to solve https://github.com/PowerShell/PowerShellEditorServices/issues/2245 without requiring any changes to PowerShell itself. It'll also enable setting breakpoints on an attach scenario for 5.1 without having to wait for the debugger to hit a break.
